src/ProductConstructionService/ProductConstructionService.Common/StateManager.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using StackExchange.Redis;
+
+namespace ProductConstructionService.Common;
+
+public interface IStateManager
+{
+    public void Set(string value);
+    public Task SetAsync(string value);
+    public bool TryGet(out string? value);
+    public Task<string?> GetAsync();
+}
+
+public class StateManager : IStateManager
+{
+    private readonly string _stateKey;
+    private readonly IConnectionMultiplexer _connection;
+
+    internal StateManager(IConnectionMultiplexer connection, string stateKey)
+    {
+        _connection = connection;
+        _stateKey = stateKey;
+    }
+
+    public void Set(string value)
+    {
+        var cache = _connection.GetDatabase();
```

Comment:
Maybe we can just have `private TSomethingHere Cache => _connection.GetDatabase()`?

Reply:
The docs state that this object is a throwaway - https://stackexchange.github.io/StackExchange.Redis/Basics#using-a-redis-database

Based on this I had it created for every operation. Don't have a strong opinion though, we can just store it

Reply:
There's this:

```
public T Property => CalledEverytime();
```
and
```
public T Property {get;} = CalledOnce();
```